src/libraries/System.Runtime.Extensions/tests/System/OperatingSystemTests.cs

```diff
@@ -47,5 +47,489 @@ public static void Clone()
+            AssertVersionChecks(OperatingSystem.IsAndroidVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("ANDROID", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsFreeBSDVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("FREEBSD", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsIOSVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("IOS", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsMacOSVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("MACOS", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsTvOSVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("TVOS", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsWatchOSVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("WATCHTV", major, minor, build, revision), false);
+            AssertVersionChecks(OperatingSystem.IsWindowsVersionAtLeast, false);
+            AssertVersionChecks((major, minor, build, revision) => OperatingSystem.IsOSPlatformVersionAtLeast("WINDOWS", major, minor, build, revision), false);
+        }
+
+        [Fact, PlatformSpecific(TestPlatforms.Linux)]
+        public static void CheckLinux()
+        {
+            Assert.False(OperatingSystem.IsBrowser());
+            Assert.False(OperatingSystem.IsOSPlatform("BROWSER"));
+            Assert.False(OperatingSystem.IsOSPlatform("browser"));
+            Assert.True(OperatingSystem.IsLinux());
+            Assert.True(OperatingSystem.IsOSPlatform("LINUX"));
+            Assert.True(OperatingSystem.IsOSPlatform("linux"));
+            Assert.False(OperatingSystem.IsFreeBSD());
+            Assert.False(OperatingSystem.IsOSPlatform("FREEBSD"));
+            Assert.False(OperatingSystem.IsOSPlatform("freebsd"));
+            Assert.False(OperatingSystem.IsAndroid());
+            Assert.False(OperatingSystem.IsOSPlatform("ANDROID"));
+            Assert.False(OperatingSystem.IsOSPlatform("android"));
+            Assert.False(OperatingSystem.IsIOS());
+            Assert.False(OperatingSystem.IsOSPlatform("IOS"));
+            Assert.False(OperatingSystem.IsOSPlatform("iOS"));
+            Assert.False(OperatingSystem.IsOSPlatform("ios"));
+            Assert.False(OperatingSystem.IsMacOS());
+            Assert.False(OperatingSystem.IsOSPlatform("MACOS"));
```

Comment:
Nit, is there some code duplication that could be collapsed, eg., stuff like
```c#
            Assert.False(OperatingSystem.IsOSPlatform("ANDROID"));
            Assert.False(OperatingSystem.IsOSPlatform("android"));
            Assert.False(OperatingSystem.IsOSPlatform("IOS"));
            Assert.False(OperatingSystem.IsOSPlatform("iOS"));
            Assert.False(OperatingSystem.IsOSPlatform("ios"));
```
could become eg
```c#
           AssertNotOSPlatform(new [] { "android", "ios", ...} );
..
          private void AssertNotOSPlatform(string[] notPlatforms)          =>
                 // assert OperatingSystem.IsOSPlatform is false for all casings
```
similarly 
```c#
            Assert.True(OperatingSystem.IsBrowser());
            Assert.False(OperatingSystem.IsLinux());
            Assert.False(OperatingSystem.IsFreeBSD());
            Assert.False(OperatingSystem.IsAndroid());
   ..
```
could maybe be
```c#
            Assert.True(OperatingSystem.IsBrowser());
            AssertOnlyOneOperatingSystemIsTrue();
```
I dunno.

Reply:
and similar for AssertVersionChecks

Reply:
@danmosemsft there was certainly a big place for improvement in the tests that I've provided. I've refactored them according to your suggestion. It looks much better now, thanks!